A JavaScript error occurred in the main process occurs anytime rotki comes or goes out of focus on the screen

## Problem Definition
A JavaScript error occurred in the main process

Uncaught Exception:
AssertionError: AssertionError
at _n (/Applications/rotki.app/Contents/Resources/app.asar/dist/background.js:70:31894)
at get tray [as tray] (/Applications/rotki.app/Contents/Resources/app.asar/dist/background.js:70:41791)
at BrowserWindow.hidden (/Applications/rotki.app/Contents/Resources/app.asar/dist/background.js:70:41570)
at BrowserWindow.emit (node:events:526:35)

![Screenshot 2024-02-09 at 15 48 08](https://github.com/rotki/rotki/assets/10906913/b27147e4-49c2-4346-b026-192ed82af796)

### System Description
App Version: 1.31.3
Frontend Version: Frontend Version:
Platform: darwin arm64 14.3.1
Electron: 27.0.1
Premium Version: 7.0.1
Build: 1701857296807

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -2,6 +2,7 @@
 Changelog
 =========
 
+* :bug:`7435` Fix the issue where, in rare cases, users receive a JavaScript error popup when minimizing the window.
 * :feature:`7357` rotki will now properly detect and decode Shutter token airdrop and delegation.
 * :feature:`-` Transactions contributing to Juicebox campaigns, such as the Free Alexey and Roman from Tornado, are now properly decoded.
 * :feature:`-` Octant epoch reward claiming transactions are now properly decoded.

--- a/frontend/app/src/electron-main/tray-manager.ts
+++ b/frontend/app/src/electron-main/tray-manager.ts
@@ -2,10 +2,8 @@ import path from 'node:path';
 import process from 'node:process';
 import { type BrowserWindow, Menu, Tray, app } from 'electron';
 import { settingsManager } from '@/electron-main/app-settings';
-import { assert } from '@/utils/assertions';
 import { checkIfDevelopment } from '@/utils/env-utils';
 import type { TrayUpdate } from '@/electron-main/ipc';
-import type { Nullable } from '@/types';
 
 type WindowProvider = () => BrowserWindow;
 const isMac = process.platform === 'darwin';
@@ -14,12 +12,7 @@ export class TrayManager {
   private readonly getWindow: WindowProvider;
   private readonly closeApp: () => void;
   private tooltip = '';
-  private _tray: Nullable<Tray> = null;
-
-  private get tray(): Tray {
-    assert(this._tray);
-    return this._tray;
-  }
+  private tray?: Tray = undefined;
 
   constructor(getWindow: WindowProvider, closeApp: () => void) {
     this.getWindow = getWindow;
@@ -63,7 +56,7 @@ export class TrayManager {
   }
 
   update({ currency, delta, percentage, up, period, netWorth }: TrayUpdate) {
-    if (!this._tray)
+    if (!this.tray)
       return;
 
     if (up === undefined) {
@@ -100,7 +93,7 @@ export class TrayManager {
 
   private setIcon(iconName: string) {
     const iconPath = path.join(TrayManager.iconPath, iconName);
-    this.tray.setImage(iconPath);
+    this.tray?.setImage(iconPath);
   }
 
   private showHide(win: BrowserWindow) {
@@ -112,21 +105,21 @@ export class TrayManager {
       app.focus();
     }
 
-    this.tray.setContextMenu(this.buildMenu(win.isVisible()));
+    this.tray?.setContextMenu(this.buildMenu(win.isVisible()));
   }
 
   private hidden = () => {
-    this.tray.setContextMenu(this.buildMenu(false, this.tooltip));
+    this.tray?.setContextMenu(this.buildMenu(false, this.tooltip));
   };
 
   private shown = () => {
-    this.tray.setContextMenu(this.buildMenu(true, this.tooltip));
+    this.tray?.setContextMenu(this.buildMenu(true, this.tooltip));
   };
 
   build() {
     const icon = isMac ? 'rotki-trayTemplate.png' : 'rotki_tray.png';
     const iconPath = path.join(TrayManager.iconPath, icon);
-    this._tray = new Tray(iconPath);
+    this.tray = new Tray(iconPath);
     this.tray.setToolTip('rotki is running');
 
     this.tray.setContextMenu(this.buildMenu(true));
@@ -135,8 +128,10 @@ export class TrayManager {
   }
 
   destroy() {
-    this._tray?.destroy();
-    this._tray = null;
+    if (this.tray) {
+      this.tray.destroy();
+      this.tray = undefined;
+    }
   }
 
   listen() {

